## 1️⃣ PREPARAÇÃO DOS DADOS

In [1]:
install.packages(c(
  "readxl", "lavaan", "semTools", "psych",
  "tidyverse", "blavaan", "bayesplot"
))

library(readxl)
library(lavaan)
library(semTools)
library(psych)
library(tidyverse)
library(blavaan)
library(bayesplot)

Instalando pacotes em 'C:/Users/Maxwell/AppData/Local/R/win-library/4.5'
(como 'lib' não foi especificado)



pacote 'readxl' desempacotado com sucesso e somas MD5 verificadas
pacote 'lavaan' desempacotado com sucesso e somas MD5 verificadas
pacote 'semTools' desempacotado com sucesso e somas MD5 verificadas
pacote 'psych' desempacotado com sucesso e somas MD5 verificadas
pacote 'tidyverse' desempacotado com sucesso e somas MD5 verificadas
pacote 'blavaan' desempacotado com sucesso e somas MD5 verificadas
pacote 'bayesplot' desempacotado com sucesso e somas MD5 verificadas

Os pacotes binários baixados estão em
	C:\Users\Maxwell\AppData\Local\Temp\RtmpoBBIEl\downloaded_packages


This is lavaan 0.6-21
lavaan is FREE software! Please report any bugs.

 

###############################################################################

This is semTools 0.5-7

All users of R (or SEM) are invited to submit functions or ideas for functions.

###############################################################################


Anexando pacote: 'psych'


Os seguintes objetos são mascarados por 'package:semTools':

    reliability, skew


O seguinte objeto é mascarado por 'package:lavaan':

    cor2cov


── Attaching core tidyverse packages ──────────────────────────────────────────────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.6
✔ forcats   1.0.1     ✔ stringr   1.6.0
✔ ggplot2   4.0.1     ✔ tibble    3.3.1
✔ lubridate 1.9.4     ✔ tidyr     1.3.2
✔ purrr     1.2.1     
── Conflicts ────────────────────────────────────────────────────────────────────────────────── tidyverse_conflicts() ──
✖ ggplot2::%+%()     masks psych::%+%()
✖ ggplot2::al

In [2]:
dados <- read_excel("dados-cp.xlsx")

In [5]:
colnames(dados) <- paste0("Q", 1:72)

## 2️⃣ ESPECIFICAÇÃO TEÓRICA DO MODELO DE AFC

In [3]:
modelo_emocional <- '
Consciencia_Emocional =~ Q48 + Q49 + Q50 + Q51 + Q52 + Q53 + Q54

Regulacao_Emocional =~ Q55 + Q56 + Q57 + Q58 + Q59 + Q60

Consciencia_Social =~ Q61 + Q62 + Q63 + Q64

Autocontrole_Emocional =~ Q65 + Q66 + Q67

Criatividade_Emocional =~ Q68 + Q69 + Q70 + Q71 + Q72
'

In [44]:
modelo_trabalho <- '
Autonomia_Variedade =~ Q6 + Q7 + Q8 + Q9 + Q10 + Q11

Signif_Identif_Feedback =~ Q12 + Q13 + Q14 + Q15 + Q16 + Q17 + Q18

Complexidade_Proc_Criativ =~ Q19 + Q20 + Q21 + Q22 + Q23 + Q24 + Q25 + Q26 + Q27

Apoio_Interdep_Interacao =~ Q28 + Q29 + Q30 + Q31 + Q32 + Q33 + Q34 + Q35 + Q36 + Q37

Exig_Fisicas_Ambiente =~ Q38 + Q39 + Q40 + Q41 + Q42 + Q43 + Q44 + Q45 + Q46 + Q47
'

## 3️⃣ AFC ROBUSTA (MLR)

In [6]:
fit_emocional <- cfa(
  modelo_emocional,
  data = dados,
  estimator = "MLR",
  std.lv = TRUE,
  missing = "fiml"
)

In [47]:
fitMeasures(fit_emocional, c(
  "chisq","df","pvalue","cfi","tli","rmsea","rmsea.ci.lower","rmsea.ci.upper","srmr"
))

chisq             df         pvalue            cfi            tli 
       774.699        265.000          0.000          0.919          0.908 
         rmsea rmsea.ci.lower rmsea.ci.upper           srmr 
         0.077          0.071          0.084          0.050

In [45]:
fit_trabalho <- cfa(
  modelo_trabalho,
  data = dados,
  estimator = "MLR",
  std.lv = TRUE,
  missing = "fiml"
)

In [48]:
fitMeasures(fit_trabalho, c(
  "chisq","df","pvalue","cfi","tli","rmsea","rmsea.ci.lower","rmsea.ci.upper","srmr"
))

chisq             df         pvalue            cfi            tli 
      3453.499        809.000          0.000          0.592          0.566 
         rmsea rmsea.ci.lower rmsea.ci.upper           srmr 
         0.101          0.097          0.104          0.113

## 5️⃣ CARGAS FATORIAIS E CONFIABILIDADE

In [8]:
standardizedSolution(fit_emocional) %>% filter(op == "=~")

lhs,op,rhs,est.std,se,z,pvalue,ci.lower,ci.upper
<chr>,<chr>,<chr>,<dbl>,<dbl>,<dbl>,<dbl>,<dbl>,<dbl>
Consciencia_Emocional,=~,Q48,0.8372264,0.02738787,30.56924,0,0.7835471,0.8909056
Consciencia_Emocional,=~,Q49,0.7798641,0.03539036,22.03606,0,0.7105003,0.8492280
Consciencia_Emocional,=~,Q50,0.7693456,0.03734382,20.60169,0,0.6961530,0.8425381
Consciencia_Emocional,=~,Q51,0.7930523,0.03664725,21.64016,0,0.7212250,0.8648796
Consciencia_Emocional,=~,Q52,0.7875459,0.03922618,20.07705,0,0.7106640,0.8644278
Consciencia_Emocional,=~,Q53,0.6945339,0.04341919,15.99601,0,0.6094339,0.7796340
Consciencia_Emocional,=~,Q54,0.7203988,0.04922080,14.63607,0,0.6239278,0.8168698
Regulacao_Emocional,=~,Q55,0.7147904,0.03908189,18.28956,0,0.6381913,0.7913895
Regulacao_Emocional,=~,Q56,0.7698734,0.03778729,20.37387,0,0.6958116,0.8439351


In [9]:
semTools::reliability(fit_emocional)

Warning message in semTools::reliability(fit_emocional):
"
The reliability() function was deprecated in 2022 and will cease to be included in future versions of semTools. See help('semTools-deprecated) for details.

It is replaced by the compRelSEM() function, which can estimate alpha and model-based reliability in an even wider variety of models and data types, with greater control in specifying the desired type of reliability coefficient (i.e., more explicitly choosing assumptions). 

The average variance extracted should never have been included because it is not a reliability coefficient. It is now available from the AVE() function."


,Consciencia_Emocional,Regulacao_Emocional,Consciencia_Social,Autocontrole_Emocional,Criatividade_Emocional
alpha,0.9093936,0.8974019,0.9315357,0.9096510,0.8984723
omega,0.9090518,0.8970471,0.9325157,0.9104219,0.8992509
omega2,0.9090518,0.8970471,0.9325157,0.9104219,0.8992509
omega3,0.9068019,0.8946099,0.9345366,0.9096524,0.9006337
avevar,0.5885939,0.5928338,0.7755733,0.7725860,0.6411947


In [10]:
semTools::compRelSEM(fit_emocional)

Consciencia_Emocional    Regulacao_Emocional     Consciencia_Social 
                 0.907                  0.895                  0.935 
Autocontrole_Emocional Criatividade_Emocional 
                 0.910                  0.901

## 6️⃣ VALIDADE DISCRIMINANTE

In [11]:
cor_lat <- inspect(fit_emocional, "cor.lv")
sqrt_ave <- sqrt(AVE(fit_emocional))

sqrt_ave > cor_lat

,Consciencia_Emocional,Regulacao_Emocional,Consciencia_Social,Autocontrole_Emocional,Criatividade_Emocional
Consciencia_Emocional,FALSE,TRUE,TRUE,TRUE,TRUE
Regulacao_Emocional,TRUE,FALSE,TRUE,TRUE,TRUE
Consciencia_Social,TRUE,TRUE,FALSE,TRUE,TRUE
Autocontrole_Emocional,TRUE,TRUE,TRUE,FALSE,TRUE
Criatividade_Emocional,TRUE,TRUE,TRUE,TRUE,FALSE


In [12]:
cor_lat

,Consciencia_Emocional,Regulacao_Emocional,Consciencia_Social,Autocontrole_Emocional,Criatividade_Emocional
Consciencia_Emocional,1.0000000,0.6060820,0.6144183,0.5618464,0.6941836
Regulacao_Emocional,0.6060820,1.0000000,0.6236851,0.5870554,0.6688011
Consciencia_Social,0.6144183,0.6236851,1.0000000,0.7422725,0.7188923
Autocontrole_Emocional,0.5618464,0.5870554,0.7422725,1.0000000,0.6709174
Criatividade_Emocional,0.6941836,0.6688011,0.7188923,0.6709174,1.0000000


In [13]:
sqrt_ave

Consciencia_Emocional    Regulacao_Emocional     Consciencia_Social 
                 0.767                  0.770                  0.881 
Autocontrole_Emocional Criatividade_Emocional 
                 0.879                  0.801

## 7️⃣ ANÁLISE BAYESIANA FATORIAL

In [14]:
fit_bayes_emocional <- bcfa(
  modelo_emocional,
  data = dados,
  burnin = 2000,
  sample = 5000,
  n.chains = 4
)


SAMPLING FOR MODEL 'stanmarg' NOW (CHAIN 1).
Chain 1: 
Chain 1: Gradient evaluation took 0.003908 seconds
Chain 1: 1000 transitions using 10 leapfrog steps per transition would take 39.08 seconds.
Chain 1: Adjust your expectations accordingly!
Chain 1: 
Chain 1: 
Chain 1: Iteration:    1 / 7000 [  0%]  (Warmup)
Chain 1: Iteration:  700 / 7000 [ 10%]  (Warmup)
Chain 1: Iteration: 1400 / 7000 [ 20%]  (Warmup)
Chain 1: Iteration: 2001 / 7000 [ 28%]  (Sampling)
Chain 1: Iteration: 2700 / 7000 [ 38%]  (Sampling)
Chain 1: Iteration: 3400 / 7000 [ 48%]  (Sampling)
Chain 1: Iteration: 4100 / 7000 [ 58%]  (Sampling)
Chain 1: Iteration: 4800 / 7000 [ 68%]  (Sampling)
Chain 1: Iteration: 5500 / 7000 [ 78%]  (Sampling)
Chain 1: Iteration: 6200 / 7000 [ 88%]  (Sampling)
Chain 1: Iteration: 6900 / 7000 [ 98%]  (Sampling)
Chain 1: Iteration: 7000 / 7000 [100%]  (Sampling)
Chain 1: 
Chain 1:  Elapsed Time: 65.015 seconds (Warm-up)
Chain 1:                154.002 seconds (Sampling)
Chain 1:           

In [15]:
summary(fit_bayes_emocional)

blavaan 0.5.10 ended normally after 5000 iterations

  Estimator                                      BAYES
  Optimization method                             MCMC
  Number of model parameters                        60

  Number of observations                           323

  Statistic                                 MargLogLik         PPP
  Value                                      -9617.061       0.000

Parameter Estimates:


Latent Variables:
                            Estimate  Post.SD pi.lower pi.upper     Rhat
  Consciencia_Emocional =~                                              
    Q48                        1.000                                    
    Q49                        1.022    0.065    0.901    1.156    1.000
    Q50                        0.947    0.061    0.831    1.071    1.000
    Q51                        0.941    0.058    0.831    1.060    1.000
    Q52                        0.947    0.058    0.839    1.066    1.000
    Q53                        0.953  

In [16]:
parameterEstimates(
  fit_bayes_emocional,
  ci = TRUE,
  level = 0.95
)

lhs,op,rhs,est
<chr>,<chr>,<chr>,<dbl>
Consciencia_Emocional,=~,Q48,1.0000000
Consciencia_Emocional,=~,Q49,1.0223564
Consciencia_Emocional,=~,Q50,0.9469482
Consciencia_Emocional,=~,Q51,0.9412162
Consciencia_Emocional,=~,Q52,0.9474950
Consciencia_Emocional,=~,Q53,0.9528871
Consciencia_Emocional,=~,Q54,0.8880050
Regulacao_Emocional,=~,Q55,1.0000000
Regulacao_Emocional,=~,Q56,0.9939564


## 8️⃣ INVARIÂNCIA DE MEDIDA

#### 🔹Gênero

In [23]:
fit_bayes_config_gender <- bcfa(
  modelo_emocional,
  data = dados,
  group = "Q1",        # exemplo: gênero
  burnin = 2000,
  sample = 5000,
  n.chains = 4
)


SAMPLING FOR MODEL 'stanmarg' NOW (CHAIN 1).
Chain 1: 
Chain 1: Gradient evaluation took 0.005145 seconds
Chain 1: 1000 transitions using 10 leapfrog steps per transition would take 51.45 seconds.
Chain 1: Adjust your expectations accordingly!
Chain 1: 
Chain 1: 
Chain 1: Iteration:    1 / 7000 [  0%]  (Warmup)
Chain 1: Iteration:  700 / 7000 [ 10%]  (Warmup)
Chain 1: Iteration: 1400 / 7000 [ 20%]  (Warmup)
Chain 1: Iteration: 2001 / 7000 [ 28%]  (Sampling)
Chain 1: Iteration: 2700 / 7000 [ 38%]  (Sampling)
Chain 1: Iteration: 3400 / 7000 [ 48%]  (Sampling)
Chain 1: Iteration: 4100 / 7000 [ 58%]  (Sampling)
Chain 1: Iteration: 4800 / 7000 [ 68%]  (Sampling)
Chain 1: Iteration: 5500 / 7000 [ 78%]  (Sampling)
Chain 1: Iteration: 6200 / 7000 [ 88%]  (Sampling)
Chain 1: Iteration: 6900 / 7000 [ 98%]  (Sampling)
Chain 1: Iteration: 7000 / 7000 [100%]  (Sampling)
Chain 1: 
Chain 1:  Elapsed Time: 231.359 seconds (Warm-up)
Chain 1:                459.158 seconds (Sampling)
Chain 1:          

In [30]:
summary(fit_bayes_config_gender)

blavaan 0.5.10 ended normally after 5000 iterations

  Estimator                                      BAYES
  Optimization method                             MCMC
  Number of model parameters                       170

  Number of observations per group:                   
    Masculino                                      150
    Feminino                                       173

  Statistic                                 MargLogLik         PPP
  Value                                     -10034.267       0.000

Parameter Estimates:



Group 1 [Masculino]:

Latent Variables:
                            Estimate  Post.SD pi.lower pi.upper     Rhat
  Consciencia_Emocional =~                                              
    Q48                        1.000                                    
    Q49                        0.969    0.099    0.787    1.175    1.000
    Q50                        0.863    0.099    0.678    1.069    1.000
    Q51                        0.935    0.095    0.

In [24]:
fit_bayes_metric_gender <- bcfa(
  modelo_emocional,
  data = dados,
  group = "Q1",
  group.equal = "loadings",
  burnin = 2000,
  sample = 5000,
  n.chains = 4
)


SAMPLING FOR MODEL 'stanmarg' NOW (CHAIN 1).
Chain 1: 
Chain 1: Gradient evaluation took 0.003631 seconds
Chain 1: 1000 transitions using 10 leapfrog steps per transition would take 36.31 seconds.
Chain 1: Adjust your expectations accordingly!
Chain 1: 
Chain 1: 
Chain 1: Iteration:    1 / 7000 [  0%]  (Warmup)
Chain 1: Iteration:  700 / 7000 [ 10%]  (Warmup)
Chain 1: Iteration: 1400 / 7000 [ 20%]  (Warmup)
Chain 1: Iteration: 2001 / 7000 [ 28%]  (Sampling)
Chain 1: Iteration: 2700 / 7000 [ 38%]  (Sampling)
Chain 1: Iteration: 3400 / 7000 [ 48%]  (Sampling)
Chain 1: Iteration: 4100 / 7000 [ 58%]  (Sampling)
Chain 1: Iteration: 4800 / 7000 [ 68%]  (Sampling)
Chain 1: Iteration: 5500 / 7000 [ 78%]  (Sampling)
Chain 1: Iteration: 6200 / 7000 [ 88%]  (Sampling)
Chain 1: Iteration: 6900 / 7000 [ 98%]  (Sampling)
Chain 1: Iteration: 7000 / 7000 [100%]  (Sampling)
Chain 1: 
Chain 1:  Elapsed Time: 189.923 seconds (Warm-up)
Chain 1:                424.82 seconds (Sampling)
Chain 1:           

In [31]:
summary(fit_bayes_metric_gender)

blavaan 0.5.10 ended normally after 5000 iterations

  Estimator                                      BAYES
  Optimization method                             MCMC
  Number of model parameters                       170
  Number of equality constraints                    20

  Number of observations per group:                   
    Masculino                                      150
    Feminino                                       173

  Statistic                                 MargLogLik         PPP
  Value                                      -9954.450       0.000

Parameter Estimates:



Group 1 [Masculino]:

Latent Variables:
                            Estimate  Post.SD pi.lower pi.upper     Rhat
  Consciencia_Emocional =~                                              
    Q48                        1.000                                    
    Q49     (.p2.)             1.013    0.066    0.887    1.148    1.000
    Q50     (.p3.)             0.968    0.060    0.855    1.091    1.

In [21]:
fit_bayes_gender <- bcfa(
  modelo_emocional,
  data = dados,
  group = "Q1",
  burnin = 2000,
  sample = 5000,
  n.chains = 4,
  group.equal = c("loadings", "intercepts")
)


SAMPLING FOR MODEL 'stanmarg' NOW (CHAIN 1).
Chain 1: 
Chain 1: Gradient evaluation took 0.003727 seconds
Chain 1: 1000 transitions using 10 leapfrog steps per transition would take 37.27 seconds.
Chain 1: Adjust your expectations accordingly!
Chain 1: 
Chain 1: 
Chain 1: Iteration:    1 / 7000 [  0%]  (Warmup)
Chain 1: Iteration:  700 / 7000 [ 10%]  (Warmup)
Chain 1: Iteration: 1400 / 7000 [ 20%]  (Warmup)
Chain 1: Iteration: 2001 / 7000 [ 28%]  (Sampling)
Chain 1: Iteration: 2700 / 7000 [ 38%]  (Sampling)
Chain 1: Iteration: 3400 / 7000 [ 48%]  (Sampling)
Chain 1: Iteration: 4100 / 7000 [ 58%]  (Sampling)
Chain 1: Iteration: 4800 / 7000 [ 68%]  (Sampling)
Chain 1: Iteration: 5500 / 7000 [ 78%]  (Sampling)
Chain 1: Iteration: 6200 / 7000 [ 88%]  (Sampling)
Chain 1: Iteration: 6900 / 7000 [ 98%]  (Sampling)
Chain 1: Iteration: 7000 / 7000 [100%]  (Sampling)
Chain 1: 
Chain 1:  Elapsed Time: 190.325 seconds (Warm-up)
Chain 1:                555.155 seconds (Sampling)
Chain 1:          

In [29]:
summary(fit_bayes_gender)

blavaan 0.5.10 ended normally after 5000 iterations

  Estimator                                      BAYES
  Optimization method                             MCMC
  Number of model parameters                       175
  Number of equality constraints                    45

  Number of observations per group:                   
    Masculino                                      150
    Feminino                                       173

  Statistic                                 MargLogLik         PPP
  Value                                      -9843.593       0.000

Parameter Estimates:



Group 1 [Masculino]:

Latent Variables:
                            Estimate  Post.SD pi.lower pi.upper     Rhat
  Consciencia_Emocional =~                                              
    Q48                        1.000                                    
    Q49     (.p2.)             1.022    0.065    0.899    1.154    1.000
    Q50     (.p3.)             0.972    0.059    0.861    1.096    1.

Foi conduzida uma análise de invariância de medida Bayesiana entre homens (n = 150) e mulheres (n = 173). O modelo configural apresentou adequada representação da estrutura fatorial em ambos os grupos, com cargas fatoriais elevadas e intervalos de credibilidade que não incluíram zero. Em seguida, o modelo métrico, no qual as cargas fatoriais foram igualadas entre os grupos, apresentou melhora no ajuste Bayesiano, conforme indicado pelo aumento do logaritmo da verossimilhança marginal. Por fim, o modelo escalar, com cargas e interceptos restritos à igualdade, manteve estabilidade das estimativas e apresentou o melhor ajuste relativo. Esses resultados fornecem evidência robusta de invariância configural, métrica e escalar do instrumento entre os gêneros, permitindo a comparação válida de médias latentes.

In [36]:
fit_bayes_gender

blavaan 0.5.10 ended normally after 5000 iterations

  Estimator                                      BAYES
  Optimization method                             MCMC
  Number of model parameters                       175
  Number of equality constraints                    45

  Number of observations per group:                   
    Masculino                                      150
    Feminino                                       173

  Statistic                                 MargLogLik         PPP
  Value                                      -9843.593       0.000

#### 🔹Liderança

In [25]:
fit_bayes_config_leader <- bcfa(
  modelo_emocional,
  data = dados,
  group = "Q4",        
  burnin = 2000,
  sample = 5000,
  n.chains = 4
)


SAMPLING FOR MODEL 'stanmarg' NOW (CHAIN 1).
Chain 1: 
Chain 1: Gradient evaluation took 0.003716 seconds
Chain 1: 1000 transitions using 10 leapfrog steps per transition would take 37.16 seconds.
Chain 1: Adjust your expectations accordingly!
Chain 1: 
Chain 1: 
Chain 1: Iteration:    1 / 7000 [  0%]  (Warmup)
Chain 1: Iteration:  700 / 7000 [ 10%]  (Warmup)
Chain 1: Iteration: 1400 / 7000 [ 20%]  (Warmup)
Chain 1: Iteration: 2001 / 7000 [ 28%]  (Sampling)
Chain 1: Iteration: 2700 / 7000 [ 38%]  (Sampling)
Chain 1: Iteration: 3400 / 7000 [ 48%]  (Sampling)
Chain 1: Iteration: 4100 / 7000 [ 58%]  (Sampling)
Chain 1: Iteration: 4800 / 7000 [ 68%]  (Sampling)
Chain 1: Iteration: 5500 / 7000 [ 78%]  (Sampling)
Chain 1: Iteration: 6200 / 7000 [ 88%]  (Sampling)
Chain 1: Iteration: 6900 / 7000 [ 98%]  (Sampling)
Chain 1: Iteration: 7000 / 7000 [100%]  (Sampling)
Chain 1: 
Chain 1:  Elapsed Time: 237.107 seconds (Warm-up)
Chain 1:                453.893 seconds (Sampling)
Chain 1:          

In [32]:
summary(fit_bayes_config_leader)

blavaan 0.5.10 ended normally after 5000 iterations

  Estimator                                      BAYES
  Optimization method                             MCMC
  Number of model parameters                       170

  Number of observations per group:                   
    Não                                            196
    Sim                                            127

  Statistic                                 MargLogLik         PPP
  Value                                     -10025.163       0.000

Parameter Estimates:



Group 1 [Não]:

Latent Variables:
                            Estimate  Post.SD pi.lower pi.upper     Rhat
  Consciencia_Emocional =~                                              
    Q48                        1.000                                    
    Q49                        1.046    0.080    0.896    1.211    1.000
    Q50                        0.881    0.073    0.745    1.030    1.000
    Q51                        0.910    0.072    0.776   

In [26]:
fit_bayes_metric <- bcfa(
  modelo_emocional,
  data = dados,
  group = "Q4",
  group.equal = "loadings",
  burnin = 2000,
  sample = 5000,
  n.chains = 4
)


SAMPLING FOR MODEL 'stanmarg' NOW (CHAIN 1).
Chain 1: 
Chain 1: Gradient evaluation took 0.005139 seconds
Chain 1: 1000 transitions using 10 leapfrog steps per transition would take 51.39 seconds.
Chain 1: Adjust your expectations accordingly!
Chain 1: 
Chain 1: 
Chain 1: Iteration:    1 / 7000 [  0%]  (Warmup)
Chain 1: Iteration:  700 / 7000 [ 10%]  (Warmup)
Chain 1: Iteration: 1400 / 7000 [ 20%]  (Warmup)
Chain 1: Iteration: 2001 / 7000 [ 28%]  (Sampling)
Chain 1: Iteration: 2700 / 7000 [ 38%]  (Sampling)
Chain 1: Iteration: 3400 / 7000 [ 48%]  (Sampling)
Chain 1: Iteration: 4100 / 7000 [ 58%]  (Sampling)
Chain 1: Iteration: 4800 / 7000 [ 68%]  (Sampling)
Chain 1: Iteration: 5500 / 7000 [ 78%]  (Sampling)
Chain 1: Iteration: 6200 / 7000 [ 88%]  (Sampling)
Chain 1: Iteration: 6900 / 7000 [ 98%]  (Sampling)
Chain 1: Iteration: 7000 / 7000 [100%]  (Sampling)
Chain 1: 
Chain 1:  Elapsed Time: 192.134 seconds (Warm-up)
Chain 1:                396.869 seconds (Sampling)
Chain 1:          

In [33]:
summary(fit_bayes_metric)

blavaan 0.5.10 ended normally after 5000 iterations

  Estimator                                      BAYES
  Optimization method                             MCMC
  Number of model parameters                       170
  Number of equality constraints                    20

  Number of observations per group:                   
    Não                                            196
    Sim                                            127

  Statistic                                 MargLogLik         PPP
  Value                                      -9945.643       0.000

Parameter Estimates:



Group 1 [Não]:

Latent Variables:
                            Estimate  Post.SD pi.lower pi.upper     Rhat
  Consciencia_Emocional =~                                              
    Q48                        1.000                                    
    Q49     (.p2.)             1.015    0.066    0.889    1.150    1.000
    Q50     (.p3.)             0.957    0.063    0.838    1.084    1.000
  

In [27]:
fit_bayes_leader <- bcfa(
  modelo_emocional,
  data = dados,
  group = "Q4",
  burnin = 2000,
  sample = 5000,
  n.chains = 4,
  group.equal = c("loadings", "intercepts")
)


SAMPLING FOR MODEL 'stanmarg' NOW (CHAIN 1).
Chain 1: 
Chain 1: Gradient evaluation took 0.004811 seconds
Chain 1: 1000 transitions using 10 leapfrog steps per transition would take 48.11 seconds.
Chain 1: Adjust your expectations accordingly!
Chain 1: 
Chain 1: 
Chain 1: Iteration:    1 / 7000 [  0%]  (Warmup)
Chain 1: Iteration:  700 / 7000 [ 10%]  (Warmup)
Chain 1: Iteration: 1400 / 7000 [ 20%]  (Warmup)
Chain 1: Iteration: 2001 / 7000 [ 28%]  (Sampling)
Chain 1: Iteration: 2700 / 7000 [ 38%]  (Sampling)
Chain 1: Iteration: 3400 / 7000 [ 48%]  (Sampling)
Chain 1: Iteration: 4100 / 7000 [ 58%]  (Sampling)
Chain 1: Iteration: 4800 / 7000 [ 68%]  (Sampling)
Chain 1: Iteration: 5500 / 7000 [ 78%]  (Sampling)
Chain 1: Iteration: 6200 / 7000 [ 88%]  (Sampling)
Chain 1: Iteration: 6900 / 7000 [ 98%]  (Sampling)
Chain 1: Iteration: 7000 / 7000 [100%]  (Sampling)
Chain 1: 
Chain 1:  Elapsed Time: 174.891 seconds (Warm-up)
Chain 1:                420.379 seconds (Sampling)
Chain 1:          

In [43]:
fit_bayes_leader

blavaan 0.5.10 ended normally after 5000 iterations

  Estimator                                      BAYES
  Optimization method                             MCMC
  Number of model parameters                       175
  Number of equality constraints                    45

  Number of observations per group:                   
    Não                                            196
    Sim                                            127

  Statistic                                 MargLogLik         PPP
  Value                                      -9832.105       0.000

In [34]:
summary(fit_bayes_leader)

blavaan 0.5.10 ended normally after 5000 iterations

  Estimator                                      BAYES
  Optimization method                             MCMC
  Number of model parameters                       175
  Number of equality constraints                    45

  Number of observations per group:                   
    Não                                            196
    Sim                                            127

  Statistic                                 MargLogLik         PPP
  Value                                      -9832.105       0.000

Parameter Estimates:



Group 1 [Não]:

Latent Variables:
                            Estimate  Post.SD pi.lower pi.upper     Rhat
  Consciencia_Emocional =~                                              
    Q48                        1.000                                    
    Q49     (.p2.)             1.024    0.066    0.900    1.159    1.000
    Q50     (.p3.)             0.952    0.061    0.835    1.076    1.000
  

Foi conduzida uma análise de invariância de medida Bayesiana para avaliar a equivalência do instrumento de inteligência emocional entre profissionais com função de liderança (n = 127) e sem função de liderança (n = 196). Inicialmente, o modelo configural apresentou adequada representação da estrutura fatorial em ambos os grupos, com cargas fatoriais elevadas e intervalos de credibilidade que não incluíram zero. Em seguida, o modelo métrico, no qual as cargas fatoriais foram restringidas à igualdade entre os grupos, apresentou melhora no ajuste Bayesiano, conforme indicado pelo aumento do logaritmo da verossimilhança marginal. Por fim, o modelo escalar, com restrição adicional dos interceptos dos itens, manteve estabilidade das estimativas e apresentou o melhor ajuste relativo. Esses resultados fornecem evidência robusta de invariância configural, métrica e escalar do instrumento entre indivíduos com e sem função de liderança, permitindo a comparação válida de médias latentes entre os grupos.

In [40]:
parameterEstimates(
  fit_bayes_leader,
  ci = TRUE,
  level = 0.95
) %>%
  filter(op == "~1")

lhs,op,rhs,block,group,label,est
<chr>,<chr>,<chr>,<int>,<int>,<chr>,<dbl>
Q48,~1,,1,1,.p66.,3.9831675
Q49,~1,,1,1,.p67.,3.7717275
Q50,~1,,1,1,.p68.,3.8169089
Q51,~1,,1,1,.p69.,4.0824878
Q52,~1,,1,1,.p70.,3.9618370
Q53,~1,,1,1,.p71.,3.8099404
Q54,~1,,1,1,.p72.,4.0902097
Q55,~1,,1,1,.p73.,3.5255858
Q56,~1,,1,1,.p74.,3.7753546


## 9️⃣ MODELO MULTIGRUPO BAYESIANO FINAL

In [38]:
parameterEstimates(
  fit_bayes_gender,
  ci = TRUE,
  level = 0.95
) %>%
  filter(op == "~1")

lhs,op,rhs,block,group,label,est
<chr>,<chr>,<chr>,<int>,<int>,<chr>,<dbl>
Q48,~1,,1,1,.p66.,3.8576557
Q49,~1,,1,1,.p67.,3.6461467
Q50,~1,,1,1,.p68.,3.7024699
Q51,~1,,1,1,.p69.,3.9574469
Q52,~1,,1,1,.p70.,3.8345217
Q53,~1,,1,1,.p71.,3.6941162
Q54,~1,,1,1,.p72.,3.9685591
Q55,~1,,1,1,.p73.,3.5755639
Q56,~1,,1,1,.p74.,3.8322427


In [41]:
parameterEstimates(
  fit_bayes_gender,
  ci = TRUE,
  level = 0.95
) %>%
  filter(op == "~1", lhs %in% c(
    "Consciencia_Emocional",
    "Regulacao_Emocional",
    "Consciencia_Social",
    "Autocontrole_Emocional",
    "Criatividade_Emocional"
  ))

lhs,op,rhs,block,group,label,est
<chr>,<chr>,<chr>,<int>,<int>,<chr>,<dbl>
Consciencia_Emocional,~1,,1,1,,0.0000000
Regulacao_Emocional,~1,,1,1,,0.0000000
Consciencia_Social,~1,,1,1,,0.0000000
Autocontrole_Emocional,~1,,1,1,,0.0000000
Criatividade_Emocional,~1,,1,1,,0.0000000
Consciencia_Emocional,~1,,2,2,,0.4111311
Regulacao_Emocional,~1,,2,2,,0.1267415
Consciencia_Social,~1,,2,2,,0.1481310
Autocontrole_Emocional,~1,,2,2,,0.3140129


In [42]:
parameterEstimates(
  fit_bayes_leader,
  ci = TRUE,
  level = 0.95
) %>%
  filter(op == "~1", lhs %in% c(
    "Consciencia_Emocional",
    "Regulacao_Emocional",
    "Consciencia_Social",
    "Autocontrole_Emocional",
    "Criatividade_Emocional"
  ))


lhs,op,rhs,block,group,label,est
<chr>,<chr>,<chr>,<int>,<int>,<chr>,<dbl>
Consciencia_Emocional,~1,,1,1,,0.0000000
Regulacao_Emocional,~1,,1,1,,0.0000000
Consciencia_Social,~1,,1,1,,0.0000000
Autocontrole_Emocional,~1,,1,1,,0.0000000
Criatividade_Emocional,~1,,1,1,,0.0000000
Consciencia_Emocional,~1,,2,2,,0.2373690
Regulacao_Emocional,~1,,2,2,,0.3043762
Consciencia_Social,~1,,2,2,,0.1904483
Autocontrole_Emocional,~1,,2,2,,0.1320373


## 🔟 COMPARAÇÃO ENTRE FATORES LATENTES

In [68]:
fatores <- c(
  "Consciencia_Emocional",
  "Regulacao_Emocional",
  "Consciencia_Social",
  "Autocontrole_Emocional",
  "Criatividade_Emocional"
)

In [69]:
comb <- combn(fatores, 2, simplify = FALSE)

In [70]:
contraste_fatores <- function(post, f1, f2) {
  delta <- post[[1]][, paste0(f1, "~1.g2")] -
           post[[1]][, paste0(f2, "~1.g2")]
  
  c(
    Media = mean(delta),
    IC2.5 = quantile(delta, .025),
    IC97.5 = quantile(delta, .975)
  )
}

In [71]:
postG <- blavInspect(fit_bayes_gender, "mcmc")

resultados_genero <- do.call(
  rbind,
  lapply(comb, function(x) contraste_fatores(postG, x[1], x[2]))
)

rownames(resultados_genero) <- sapply(comb, paste, collapse = " - ")
resultados_genero

,Media,IC2.5.2.5%,IC97.5.97.5%
Consciencia_Emocional - Regulacao_Emocional,0.28361950,0.07828839,0.49362388
Consciencia_Emocional - Consciencia_Social,0.26266816,0.06044979,0.45870541
Consciencia_Emocional - Autocontrole_Emocional,0.09693114,-0.13957232,0.32828950
Consciencia_Emocional - Criatividade_Emocional,0.15689357,-0.02854064,0.34604807
Regulacao_Emocional - Consciencia_Social,-0.02095134,-0.21681097,0.17090659
Regulacao_Emocional - Autocontrole_Emocional,-0.18668837,-0.40735008,0.03937597
Regulacao_Emocional - Criatividade_Emocional,-0.12672593,-0.31596207,0.06421700
Consciencia_Social - Autocontrole_Emocional,-0.16573702,-0.35008907,0.01568614
Consciencia_Social - Criatividade_Emocional,-0.10577459,-0.27027600,0.05781885
Autocontrole_Emocional - Criatividade_Emocional,0.05996243,-0.15287317,0.26708482


In [72]:
postL <- blavInspect(fit_bayes_leader, "mcmc")

resultados_lideranca <- do.call(
  rbind,
  lapply(comb, function(x) contraste_fatores(postL, x[1], x[2]))
)

rownames(resultados_lideranca) <- sapply(comb, paste, collapse = " - ")
resultados_lideranca

,Media,IC2.5.2.5%,IC97.5.97.5%
Consciencia_Emocional - Regulacao_Emocional,-0.06596774,-0.28884415,0.14748613
Consciencia_Emocional - Consciencia_Social,0.04630571,-0.14901031,0.24551132
Consciencia_Emocional - Autocontrole_Emocional,0.10610441,-0.12209253,0.33895944
Consciencia_Emocional - Criatividade_Emocional,-0.04215001,-0.22181511,0.13552809
Regulacao_Emocional - Consciencia_Social,0.11227345,-0.08442194,0.31136395
Regulacao_Emocional - Autocontrole_Emocional,0.17207215,-0.06048697,0.39790987
Regulacao_Emocional - Criatividade_Emocional,0.02381773,-0.16309086,0.21382300
Consciencia_Social - Autocontrole_Emocional,0.05979870,-0.12309329,0.24392000
Consciencia_Social - Criatividade_Emocional,-0.08845572,-0.25532149,0.07845536
Autocontrole_Emocional - Criatividade_Emocional,-0.14825442,-0.35829645,0.06525988


In [73]:
diff_diff_fatores <- function(postG, postL, f1, f2) {
  
  # diferença entre fatores – mulheres
  delta_mulheres <- postG[[1]][, paste0(f1, "~1.g2")] -
                    postG[[1]][, paste0(f2, "~1.g2")]
  
  # diferença entre fatores – líderes
  delta_lideres <- postL[[1]][, paste0(f1, "~1.g2")] -
                   postL[[1]][, paste0(f2, "~1.g2")]
  
  # diferença das diferenças
  diff_diff <- delta_mulheres - delta_lideres
  
  c(
    Media = mean(diff_diff),
    IC2.5 = quantile(diff_diff, .025),
    IC97.5 = quantile(diff_diff, .975)
  )
}

In [74]:
fatores <- c(
  "Consciencia_Emocional",
  "Regulacao_Emocional",
  "Consciencia_Social",
  "Autocontrole_Emocional",
  "Criatividade_Emocional"
)

comb <- combn(fatores, 2, simplify = FALSE)

In [75]:
postG <- blavInspect(fit_bayes_gender, "mcmc")
postL <- blavInspect(fit_bayes_leader, "mcmc")

resultados_diffdiff <- do.call(
  rbind,
  lapply(comb, function(x)
    diff_diff_fatores(postG, postL, x[1], x[2]))
)

rownames(resultados_diffdiff) <- sapply(comb, paste, collapse = " − ")

resultados_diffdiff

,Media,IC2.5.2.5%,IC97.5.97.5%
Consciencia_Emocional − Regulacao_Emocional,0.349587245,0.04713554,0.65958184
Consciencia_Emocional − Consciencia_Social,0.216362451,-0.06259912,0.49605138
Consciencia_Emocional − Autocontrole_Emocional,-0.009173272,-0.34744898,0.32323485
Consciencia_Emocional − Criatividade_Emocional,0.199043581,-0.05481080,0.45663863
Regulacao_Emocional − Consciencia_Social,-0.133224794,-0.40482626,0.14273136
Regulacao_Emocional − Autocontrole_Emocional,-0.358760517,-0.67613651,-0.03561163
Regulacao_Emocional − Criatividade_Emocional,-0.150543664,-0.41304562,0.12453513
Consciencia_Social − Autocontrole_Emocional,-0.225535723,-0.47968001,0.02899927
Consciencia_Social − Criatividade_Emocional,-0.017318870,-0.25245790,0.22132526
Autocontrole_Emocional − Criatividade_Emocional,0.208216853,-0.09220603,0.50358290


In [77]:
resultados_diffdiff <- as.data.frame(resultados_diffdiff)

resultados_diffdiff$Robusto <-
  !(resultados_diffdiff$IC2.5 < 0 & resultados_diffdiff$IC97.5 > 0)

resultados_diffdiff

,Media,IC2.5.2.5%,IC97.5.97.5%,Robusto
,<dbl>,<dbl>,<dbl>,<lgl>
Consciencia_Emocional − Regulacao_Emocional,0.349587245,0.04713554,0.65958184,TRUE
Consciencia_Emocional − Consciencia_Social,0.216362451,-0.06259912,0.49605138,FALSE
Consciencia_Emocional − Autocontrole_Emocional,-0.009173272,-0.34744898,0.32323485,FALSE
Consciencia_Emocional − Criatividade_Emocional,0.199043581,-0.05481080,0.45663863,FALSE
Regulacao_Emocional − Consciencia_Social,-0.133224794,-0.40482626,0.14273136,FALSE
Regulacao_Emocional − Autocontrole_Emocional,-0.358760517,-0.67613651,-0.03561163,TRUE
Regulacao_Emocional − Criatividade_Emocional,-0.150543664,-0.41304562,0.12453513,FALSE
Consciencia_Social − Autocontrole_Emocional,-0.225535723,-0.47968001,0.02899927,FALSE
Consciencia_Social − Criatividade_Emocional,-0.017318870,-0.25245790,0.22132526,FALSE
